packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.ct.tsx

```diff
@@ -1,48 +1,166 @@
-  await expect(component).toHaveScreenshot("invalid.png");
+  expect(accessibilityScanResults.violations).toEqual([]);
+});
+
+const disabledTestCases = [{ name: "unchecked" }, { name: "checked", modelValue: true }];
+
+for (const testCase of disabledTestCases) {
+  test(`should render disabled (${testCase.name})`, async ({ mount, makeAxeBuilder }) => {
+    // ARRANGE
+    const component = await mount(
+      <div style="display: grid; width: max-content;">
+        <OnyxSwitch label={`Disabled ${testCase.name}`} modelValue={testCase.modelValue} disabled />
+        <OnyxSwitch label="Hover" modelValue={testCase.modelValue} disabled />
+        <OnyxSwitch label="Focus visible" modelValue={testCase.modelValue} disabled />
+      </div>,
+    );
+
+    // ACT
+    const accessibilityScanResults = await makeAxeBuilder().analyze();
+
+    // ASSERT
+    expect(accessibilityScanResults.violations).toEqual([]);
+
+    const switches = await component.getByRole("checkbox").all();
+
+    // ACT
+    await switches[1].hover();
+    await switches[2].focus();
+
+    // ASSERT
+    for (const onyxSwitch of switches) {
+      await expect(onyxSwitch).toBeDisabled();
+    }
+  });
+}
+
+const invalidTestCases = [
+  { name: "unchecked" },
+  { name: "unchecked (disabled)", disabled: true },
+  { name: "indeterminate (disabled)", indeterminate: true, disabled: true },
+];
+for (const testCase of invalidTestCases) {
+  test(`should render invalid (${testCase.name})`, async ({ mount, makeAxeBuilder }) => {
+    // ARRANGE
+    const component = await mount(
+      <div style="display: grid; width: max-content;">
+        <OnyxSwitch label={`Invalid ${testCase.name}`} disabled={testCase.disabled} required />
+        <OnyxSwitch label="Hover" disabled={testCase.disabled} required />
+        <OnyxSwitch label="Focus visible" disabled={testCase.disabled} required />
+      </div>,
+    );
+
+    // ACT
+    const accessibilityScanResults = await makeAxeBuilder().analyze();
+
+    // ASSERT
+    expect(accessibilityScanResults.violations).toEqual([]);
+
+    const switches = await component.getByRole("checkbox").all();
+
+    // invalid only shows if checkbox is touched
+    for (const onyxSwitch of switches) {
+      await onyxSwitch.focus();
+      await onyxSwitch.blur();
+    }
+
+    // ACT
+    await switches[1].hover();
+    await switches[2].focus();
+  });
+}
+
+test("should have aria-label if label is hidden", async ({ mount, makeAxeBuilder }) => {
+  // ARRANGE
+  const component = await mount(<OnyxSwitch label="Test label" hideLabel />);
 
   // ACT
   const accessibilityScanResults = await makeAxeBuilder().analyze();
 
   // ASSERT
   expect(accessibilityScanResults.violations).toEqual([]);
+
+  // ASSERT
+  await expect(component).not.toContainText("Test label");
+  await expect(component.getByLabel("Test label")).toBeAttached();
 });
+
+const STATES = {
+  state: ["default", "disabled", "required", "optional"],
```

Comment:
Looking at Figma, I don't think a switch should support required/optional since its used for toggling features on/off so it can not really be required. That would be a checkbox then.

That would also mean we can remove the required property

Reply:
Well this made me think it supports required/optional feature
![image](https://github.com/SchwarzIT/onyx/assets/60856270/823bf734-ef1c-472c-bbe0-fe0ea57116f8)

Reply:
😮 
Could you please clarify this with @jannick-ux?

Reply:
It is not included in the main component variants tho: https://www.figma.com/file/YfEUBOHk4J4nYrk04geswG/onyx---Design-System-Figma-Library?type=design&node-id=711-15598&mode=dev

Reply:
Alright so nothing to change here, right?